ui/raidboss/emulator/EmulatorCommon.ts

```diff
@@ -0,0 +1,127 @@
+import { LocaleNetRegex } from '../../../resources/translations';
+import NetRegexes from '../../../resources/netregexes';
+
+// Disable ban-types for cloneData as it needs to work on raw objects for performance reasons.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DataType = { [key: string]: any } | null;
+
+export default class EmulatorCommon {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  static cloneData(data: DataType, exclude = ['options', 'party']): DataType {
+    let ret: DataType;
+    if (Array.isArray(data))
+      ret = [];
+    else
+      ret = {};
```

Comment:
This is a drive-by review of old code, but is this actually possible? I don't think `DataType` or `Data` can be an array.  Feel free to leave a TODO for later if you don't want to minimize behavior changes in this patch.

Reply:
Checking usages for this, I don't think it can be an array anymore. This helper used to be used in a few spots in the UI portion where it could be an array, if I remember correctly, and that's why it's there? Either way, I'm comfortable removing it. Removed for next commit.